### DISCUSSION TITLE
How to enable livewire for dynamically/XHR loaded components

### DISCUSSION THREAD

**Discussion by damms005** [livewire/livewire#3981] [1. Help] [answered]
`window.livewire.rescan()` used to be the got-to solution for livewire components loaded via ajax/xhr/jQuery, e.g. 

```
$(selector).load(url_to_a_livewire_component);
```
However, loading such components now throws error: `Uncaught SyntaxError: Unexpected identifier`.

To reproduce this, simply:
1. Load a normal blade file
2. Trigger an action (like click a button/link) that loads a route/action that returns a livewire page/component
3. Observe that the livewire page is "dead" (it's just a lifeless html page)

So, question is what is the go-to solution for giving life to livewire components loaded via XHR/Ajax?

**Comment by joshhanley** [comment, marked as answer]
@damms005 can you confirm what patch version of Livewire you are currently on and what patch version used to work?

The rescan function hasn't been changed in about 13 months, so it should all still be working as expected.

If you could share your code the demonstrates the loading and rescanning as it's implemented would also be helpful.

Hope this helps!

**Reply by damms005** [reply]
Yes. Thanks. Something was fundamentally wrong with my workflow

**Reply by joshhanley** [reply]
No worries 🙂
